api/resources_portal/test/views/test_material_request_issue.py

```diff
@@ -0,0 +1,266 @@
+        self.url = reverse("material-requests-issues-list", args=[self.request.id])
+
+        self.user = self.request.requester
+        self.user_without_perms = UserFactory()
+
+    def test_post_request_with_no_data_fails(self):
+        self.client.force_authenticate(user=self.user)
+        response = self.client.post(self.url, {})
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_post_request_with_valid_data_succeeds(self):
+        self.request.status = "FULFILLED"
+        self.request.save()
+
+        self.client.force_authenticate(user=self.user)
+
+        OrganizationUserSetting.objects.get_or_create(
+            user=self.sharer, organization=self.request.material.organization
+        )
+
+        response = self.client.post(self.url, self.request_issue_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_201_CREATED)
+
+        # Make sure that the request got moved back to INFULFILLMENT
+        self.request.refresh_from_db()
+        self.assertEqual(self.request.status, "IN_FULFILLMENT")
+
+        self.assertEqual(
+            len(
+                Notification.objects.filter(
+                    notification_type="REQUEST_ISSUE_OPENED", email=self.sharer.email
+                )
+            ),
+            1,
+        )
+
+    def test_post_request_unfulfilled_error(self):
+        self.client.force_authenticate(user=self.user)
+
+        OrganizationUserSetting.objects.get_or_create(
+            user=self.sharer, organization=self.request.material.organization
+        )
+
+        response = self.client.post(self.url, self.request_issue_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.json()["reason"], REQUEST_UNFULFILLED_ERROR)
+
+    def test_post_request_archived_error(self):
+        # If status is not FULFILLED then we get a different error.
+        self.request.status = "FULFILLED"
+        self.request.save()
+        self.request.material.is_archived = True
+        self.request.material.save()
+
+        self.client.force_authenticate(user=self.user)
+
+        OrganizationUserSetting.objects.get_or_create(
+            user=self.sharer, organization=self.request.material.organization
+        )
+
+        response = self.client.post(self.url, self.request_issue_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        self.assertEqual(response.json()["reason"], MATERIAL_ARCHIVED_ERROR)
+
+    def test_post_request_from_unauthenticated_forbidden(self):
+        # If status is not FULFILLED then we get a different error.
+        self.request.status = "FULFILLED"
+        self.request.save()
+        self.client.force_authenticate(user=None)
+        response = self.client.post(self.url, self.request_issue_data, format="json")
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
```

Comment:
I merged my PR after the tests for this ran I think. That PR implements token auth, which changes all the unauthenticated responses from `HTTP_403_FORBIDDEN` to `HTTP_401_UNAUTHORIZED`. This may break when you push new changes and the tests run again because of that.